blockchain/stake/internal/tickettreap/benchmark_test.go

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2016 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package tickettreap
+
+import (
+	"crypto/sha256"
+	"sync"
+	"testing"
+)
+
+// numTicketKeys is the number of keys to generate for use in the benchmarks.
+const numTicketKeys = 42500
+
+var (
+	// generatedTicketKeys is used to store ticket keys generated for use
+	// in the benchmarks so that they only need to be generatd once for all
+	// benchmarks that use them.
+	genTicketKeysLock   sync.Mutex
+	generatedTicketKeys []Key
+)
+
+// genTicketKeys generates and returns 'numTicketKeys' along with memoizing them
+// so that future calls return the cached data.
+func genTicketKeys() []Key {
+	// Return generated keys if already done.
+	genTicketKeysLock.Lock()
+	defer genTicketKeysLock.Unlock()
+	if generatedTicketKeys != nil {
+		return generatedTicketKeys
+	}
+
+	// Generate the keys and cache them for future invocations.
+	ticketKeys := make([]Key, 0, numTicketKeys)
+	for i := 0; i < numTicketKeys; i++ {
+		key := Key(sha256.Sum256(serializeUint32(uint32(i))))
+		ticketKeys = append(ticketKeys, key)
+	}
+	generatedTicketKeys = ticketKeys
+	return ticketKeys
+}
+
+// BenchmarkMutableCopy benchmarks how long it takes to copy a mutable treap
+// to another one when it contains 'numTicketKeys' entries.
+func BenchmarkMutableCopy(b *testing.B) {
+	// Populate mutable treap with a bunch of key/value pairs.
+	testTreap := NewMutable()
+	ticketKeys := genTicketKeys()
+	for j := 0; j < len(ticketKeys)-5; j++ {
```

Comment:
Oops.  That was copy pasta from when I was benchmarking removing 5 votes.  Fix them all.